packages/common/react-components/src/components/List/List.tsx

```diff
@@ -0,0 +1,270 @@
+  variant?: ListVariant;
+  onDragEnd?: ComponentPropsWithoutRef<typeof DndContext>['onDragEnd'];
+  listItemIds?: string[];
+  slots?: SharedListSlots;
+}
+
+interface DraggableListProps extends Omit<SharedListProps, 'onDragEnd' | 'listItemIds' | 'variant' | 'slots'> {
+  onDragEnd: Exclude<SharedListProps['onDragEnd'], undefined>;
+  listItemIds: Exclude<SharedListProps['listItemIds'], undefined>;
+  variant: 'ordered-draggable';
+  slots?: DraggableListSlots;
+}
+
+type ListProps = SharedListProps | DraggableListProps;
+
+interface ListItemData {
+  id: string;
+  selected?: CheckboxProps['checked'];
+}
+
+interface ListItemSlots {
+  root?: Omit<ComponentPropsWithRef<'li'>, 'id' | 'children'>;
+  dragHandle?: ComponentPropsWithoutRef<typeof ListItemDragHandle>;
+}
+
+interface ListItemProps extends Omit<ListItemData, 'id'> {
+  children?: ReactNode;
+  onSelectedChange?: CheckboxProps['onCheckedChange'];
+  defaultSelected?: CheckboxProps['defaultChecked'];
+  slots?: ListItemSlots;
+  id?: string;
+}
+
+// COLLECTION
+
+type ListItemElement = React.ElementRef<typeof Primitive.li>;
+
+const [Collection, useListCollection, createCollectionScope] = createCollection<ListItemElement, ListItemData>(
+  LIST_NAME
+);
+
+// LIST
+
+const [createListContext, createListScope] = createContextScope(LIST_NAME, [createCollectionScope]);
+
+type ListContextValue = Pick<ListProps, 'selectable' | 'variant'>;
+
+const [ListProvider, useListContext] = createListContext<ListContextValue>(LIST_NAME);
+
+const List: FC<ListProps> = (props: ScopedProps<ListProps>) => {
+  const { __scopeSelect, variant = 'ordered', selectable = false, children, slots = {} } = props;
+  const ListRoot = variant === 'ordered' || variant === 'ordered-draggable' ? Primitive.ol : Primitive.ul;
+
+  return (
+    <ListRoot
+      {...(selectable && { role: 'listbox', 'aria-multiselectable': true })}
+      {...slots.root}
+      aria-labelledby={props.labelId}
+    >
+      <ListProvider
+        {...{
+          scope: __scopeSelect,
+          variant,
+          selectable
+        }}
+      >
+        <Collection.Provider scope={__scopeSelect}>
+          {variant === 'ordered-draggable' ? (
+            <DndContext onDragEnd={(props as DraggableListProps).onDragEnd} modifiers={[restrictToVerticalAxis]}>
+              <SortableContext items={(props as DraggableListProps).listItemIds}>{children}</SortableContext>
+            </DndContext>
+          ) : (
+            <>{children}</>
+          )}
+        </Collection.Provider>
+      </ListProvider>
+    </ListRoot>
+  );
+};
+
+List.displayName = LIST_NAME;
+
+// LIST ITEM
+
+const [createListItemContext, createListItemScope] = createContextScope(LIST_ITEM_NAME, []);
+
+type ListItemContextValue = { headingId: string };
+
+const [ListItemProvider, useListItemContext] = createListItemContext<ListItemContextValue>(LIST_ITEM_NAME);
+
+const ListItemEndcap = ({ children, className, ...props }: ComponentPropsWithoutRef<'div'>) => {
```

Comment:
this isn't an obvious name to me? vs, say, ListItemCheckbox, ListItemIcon and ListItemMenu (i.e., a bit more specialize)

Reply:
I’m concerned about specializing too early if for instance we need an element with all the same styles but the content happens to be a Dialog (i.e. it would host the Dialog’s trigger).

I notice Radix usually names element that just provide a surface like this something abstract, e.g. `Item` (taken in this context), `Action`, `Content`. I hoped `Endcap` would get the point across that it should just be used at the ends, but I'm amenable to a different name that doesn’t imply a specific content.

Reply:
Come to think of it, you’d been calling this ‘gutter’, so maybe `Gutter` would be more apt?